### PR TITLE
Added pldebugger to all versions.

### LINKED ISSUE #236: Enable Debugging
I'm trying to enable / install http://www.pgadmin.org/docs/1.8/debugger.html edb-debugger plugin and all documented methods do not work with the postgres.app. Any clue on how to get this working?

Thanks in advance!

### PR DESCRIPTION
**Pull request by jasiek**
Closes #236, #341.

### CODE REVIEW

**Comment by jakob** [comment]
Awesome, thank you very much. I'll merge this as soon as I've tested it.

Could you explain what the config patch is good for? It wasn't immediately clear to me.

It also looks like the link to the pldebugger docs points to a git repo. Is that the "official" documentation for pldebugger or is there a better link?

**Comment by jasiek** [comment]
Sure thing.

The patch adds the shared object to be preloaded during server start. I've tested it without that directive, and it appears to be working, but the source code seems to allocate some shared memory on load so who knows - it's pointed out in the doc for pldebugger, so I think it needs to remain there.

As for the documentation, yes, that appears to be the canonical place for documentation for this module. It hasn't been heavily modified in a while which leads me to think it's stable and can be used.

**Comment by jakob** [comment]
Oh, I see, so the patch is changing the default PostgreSQL config file? I'm afraid I can't merge that.

Including binaries so they are there when you need them is a good idea. Loading them by default is not. We don't activate any extensions by default, not even those in contrib/. And with something as invasive as the debugger, there is just too  much risk that it'll cause issues for some users that would be very hard to diagnose.

Also, the link in the readme seems to point to a 4 year old mirror, would probably be better to link to the official repo.

Maybe we should also link to https://www.pgadmin.org/docs/pgadmin4/latest/debugger.html? It looks like a good introduction for using the debugger. The only problem is that they use a different path for `shared_preload_libraries`, so I understand why you wanted to patch the config file.

A possible option would be to patch the config file in a way that we just add a commented line with instructions to uncomment it to load the debugger. If we do add a line like that, we should probably add it after the default line `#shared_preload_libraries = ''		# (change requires restart)`.

**Comment by jasiek** [comment]
Cool, I can add a commented-out directive to preload the extension.

The link is referring to this repository (https://git.postgresql.org/gitweb/?p=pldebugger.git) which seems to be up-to-date. Is this the link you were referring to?

Adding a link to pgadmin/debugger is a good idea, I'll do that. Regarding the location of the extension - the extension ends up in $libdir, so that's where it should be loaded from. I'm not sure why they are pointing at a different path.

**Comment by jakob** [comment]
I was talking about this link in README.md: `[pldebugger](https://github.com/soycacan/pldebugger)`

The EDB installer probably installs the extension in a different directory, which is probably why pgAdmin mentions a different path. We should probably tell whoever maintains the pgAdmin docs that the path they tell people to use doesn't match what the debugger readme states...

**Comment by jasiek** [comment]
Gotcha.

What do you think about DRY-ing the pldebugger part of the makefile to a different file, perhaps in a new directory, `common/` and then including it from each `src-*` makefile?

**Comment by jakob** [comment]
I'd prefer to keep the makefiles independent. At some point we'll have to change something for a new version, and then we'll end up with separate sub-makefiles, or lots of flags everywhere. I like the fact that there's a single makefile for each major version of PostgreSQL, which is pretty straightforward to understand.

**Comment by jakob** [comment]
I made a suggestion to improve the pgAdmin4 docs: https://redmine.postgresql.org/issues/5503

**Comment by jasiek** [comment]
OK, this should be ready for testing now.

File: README.md [hunk 1]
```diff
@@ -138,6 +139,22 @@ Postgres.app also includes useful command line utilities (note: this list may be
 
 See [the documentation](http://postgresapp.com/documentation) for more info.
 
+## Using the debugger
+
+First, you'll need to adjust the configuration file (`postgresql.conf`) to preload the debugger extension. Find the following line and uncomment it:
+
+```
+# shared_preload_libraries = '$libdir/plugin_debugger'
```
Comment (tbussmann) [inline README.md:147]: Our `dynamic_library_path` defaults to `$libdir`, so there should be no need to specify that.

```suggestion
# shared_preload_libraries = 'plugin_debugger'
```

**Comment by tbussmann** [comment]
Actually I'm still not sure if we should alter the default conf file - even the official contrib modules are not 'advertised' in the conf file. If we start doing so, we should also add notes on how to enable other extensions we bundle that require loading in `shared_preload_libraries`: The following come to my mind but there may be more: `pg_prewarm`, `passwordcheck`, `auth_delay`, `pg_stat_statements`, `auto_explain`.
Further, I'm not to happy about the fact that we need to build pldebugger from git HEAD. But the project does not seem to use versioned releases, the last tag is from 2017 so there don't seem to be a better option.

**Comment by jakob** [comment]
@tbussmann Good point. We've always shipped the default config file, so probably we should just continue doing that. And considering that we don't add instructions for extensions in `contrib`, you are right, it would be weird to add instructions for pldebugger.

**Comment by jasiek** [comment]
Updated.

**Comment by jakob** [comment]
Thanks a lot! The debugger seems to work fine in my tests.

It's going to be included in todays Postgres.app releases, which I hope to release later today once they are all notarized.

**Comment by jasiek** [comment]
Excellent!

**Pull request merged**